### PR TITLE
refactor: Saga camera permissions

### PR DESCRIPTION
**Pull request by tuliomir**
### Acceptance Criteria
- Camera permissions should be requested through saga instead of directly on a screen
- "Register Token" screen should go directly to "Manual Input" when camera permission is blocked

### Security Checklist
- [X] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

File: src/components/QRCodeReader.js [hunk 1]
```diff
@@ -62,8 +62,8 @@ export default ({
     setIsFocusedScreen(true);
 
     return () => {
-      focusEvent.remove();
-      blurEvent.remove();
+      focusEvent();
+      blurEvent();
```
Comment (tuliomir) [inline src/components/QRCodeReader.js:66]: React Navigation v5.x event listeners do not have the `remove` method, and can be removed by calling them directly. ( See [API Docs](https://reactnavigation.org/docs/5.x/navigation-events#navigationaddlistener) )

File: src/App.js [hunk 2]
```diff
@@ -163,16 +202,63 @@ const CreateTokenStack = () => {
...
+   * @returns {string} Route name
+   */
+  const decideRouteByCameraAvailablity = (cameraStatus) => {
+    switch (isCameraAvailable) {
+      case true:
+        return 'RegisterToken';
+      case false:
+        return 'RegisterTokenManual';
+      default:
+        return 'RegisterCameraPermissionScreen';
```
Comment (alexruzenhack) [inline src/App.js:222]: ❓ Shouldn't it be just `'CameraPermissionScreen'`?

File: src/sagas/permissions.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+ * For a complete description on all possible permition statuses,
+ * @see https://github.com/zoontek/react-native-permissions#permissions-statuses
+ */
+export function* requestCameraPermissions() {
+  const checkStatus = async (status) => {
+    switch (status) {
+      case RESULTS.GRANTED:
+        return true;
+      case RESULTS.DENIED: // The permission has not been requested / is denied but requestable
+        return checkStatus(await request(platformCameraPermission));
```
Comment (alexruzenhack) [inline src/sagas/permissions.js:25]: ❓ Can it fall in infinity loop? Have you tested the case in which the used denied the permission over and over again?

**Review by alexruzenhack — COMMENTED** [review]
There is a possible bug in `src/sagas/permission`, I think it worths a double check.

File: src/sagas/permissions.js (hunk 3 above)
Comment (tuliomir) [reply src/sagas/permissions.js:25]: Actually this implementation fixes an infinity loop that happened before it. The simplified permissions flow works like this:
- If the user has granted unlimited permission to the application, the status returned by `check()` is `GRANTED`
- In all other cases, the status is `DENIED`.

Upon `request()`, the result can then be either `GRANTED` or `BLOCKED`. The `DENIED` is never repeated.
So, there is no loop condition unless the `react-native-permissions` lib introduces a breaking change.

Comment (alexruzenhack) [reply src/sagas/permissions.js:25]: Great!

**Review by alexruzenhack — APPROVED** [review]

**Review by andreabadesso — APPROVED** [review]

**Pull request merged**